pkgs/development/libraries/openmpi/default.nix

```diff
@@ -64,12 +76,20 @@ in stdenv.mkDerivation rec {
       "--enable-mpi-cxx"
     ] ++ lib.optional enableSGE "--with-sge"
     ++ lib.optional enablePrefix "--enable-mpirun-prefix-by-default"
-    # TODO: add UCX support, which is recommended to use with cuda for the most robust OpenMPI build
-    # https://github.com/openucx/ucx
-    # https://www.open-mpi.org/faq/?category=buildcuda
-    ++ lib.optionals cudaSupport [ "--with-cuda=${cudatoolkit_joined}" "--enable-dlopen" ]
-    ++ lib.optionals fabricSupport [ "--with-psm2=${lib.getDev libpsm2}" "--with-libfabric=${lib.getDev libfabric}" ]
-    ;
+    ++ lib.optionals cudaSupport [
+      "--with-cuda=${cudatoolkit_joined}"
+      "--enable-dlopen"
+    ] ++ lib.optionals rocmSupport [
+      "--with-rocm=${rocm}"
+    ] ++ lib.optionals (cudaSupport || rocmSupport) [
+      "--with-ucx=${lib.getDev ucx}" "--with-ucx-libdir=${ucx}/lib"
+      "--with-ucc=${lib.getDev ucc}" "--with-ucc-libdir=${ucc}/lib"
```

Comment:
I remember having to specify ``--with-libdir`` because it specifically did not do that for some reason.
I'll investigate. Could have been openmpi-5 only.